### ISSUE TITLE
Rubocop checks hashes but not function arguments!?

### ISSUE THREAD

**Issue by paulkogel** [brigade/haml-lint#132]
```Ruby
= link_to '#fixme'
```
passes whereas
```Ruby
= link_to '#fixme', { class: 'teaser-house-link' }
```
triggers `[W] RuboCop: Style/StringLiterals: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.`

It seems like not my entire Ruby code is checked by Rubocop. I'd have expected everything behind `=` to be checked by Rubocop.

Rubocop config inside `.haml-lint.yml`:

```yaml
Rubocop:
  enabled: true
  ignored_cops:
    - Lint/BlockAlignment
    - Lint/EndAlignment
    - Lint/Void
    - Metrics/LineLength
    - Style/AlignParameters
    - Style/BlockNesting
    - Style/ElseAlignment
    - Style/FileName
    - Style/FinalNewline
    - Style/FrozenStringLiteralComment
    - Style/IfUnlessModifier
    - Style/IndentationWidth
    - Style/Next
    - Style/TrailingBlankLines
    - Style/TrailingWhitespace
    - Style/WhileUntilModifier
  severity: error
```

**Comment by sds** [comment]
Hey @paulkogel,

This is a bug with RuboCop (reproduced with 0.39.0):

```yaml
# .rubocop.yml
Style/StringLiterals:
  EnforcedStyle: double_quotes
  Enabled: true
```

```ruby
# test.rb
link_to '#fixme'
```

```
$ rubocop test.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

Please file a report on the RuboCop project. Thanks!

**Comment by paulkogel** [comment]
Thanks, I just did:
https://github.com/bbatsov/rubocop/issues/3037
the hash (`#`) character is actually causing the problem. it has nothing to do whether it's a normal function argument or inside a Ruby hash (`{...}`).
